processors/parsed_attribte.go

```diff
@@ -0,0 +1,71 @@
+package processors
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type contextKey struct{}
+
+// FromContext returns ParsedTracesData and updated context
+func FromContext(ctx context.Context) (context.Context, *ParsedTracesData) {
+	sAttr, ok := ctx.Value(contextKey{}).(*ParsedTracesData)
+	if ok {
+		return ctx, sAttr
+	}
+	ptd := &ParsedTracesData{
+		spanAttributeMap: map[pdata.SpanID]*ParsedSpanData{},
+	}
+	ctx = context.WithValue(ctx, contextKey{}, ptd)
+	return ctx, ptd
+}
+
+// ParsedTracesData
+type ParsedTracesData struct {
+	spanAttributeMap map[pdata.SpanID]*ParsedSpanData
+}
+
+// GetParsedSpanData returns ParsedSpanData for a given span.
+func (p *ParsedTracesData) GetParsedSpanData(spanID pdata.SpanID) *ParsedSpanData {
+	pSpanData, ok := p.spanAttributeMap[spanID]
+	if !ok {
+		pSpanData = &ParsedSpanData{
+			parsedAttributes: map[string]*ParsedAttribute{},
+		}
+		p.spanAttributeMap[spanID] = pSpanData
```

Comment:
Is spanID enough? I'd rather use traceID+spanID as spanID is unique in a trace but not generally unique.

Reply:
There is definitely some probability to have a collision, I have changed it to Span struct as key - it's even more idiomatic.